flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/active/ActiveResourceManagerFactoryTest.java

```diff
@@ -47,19 +46,19 @@
     private static final MemorySize TOTAL_FLINK_SIZE = MemorySize.ofMebiBytes(2 * 1024);
     private static final MemorySize TOTAL_PROCESS_SIZE = MemorySize.ofMebiBytes(3 * 1024);
 
-    private static Map<String, String> systemEnv;
+    private static Properties systemProperties;
 
     @BeforeClass
     public static void setupClass() {
-        systemEnv = System.getenv();
+        systemProperties = System.getProperties();
         System.clearProperty("flink.tests.disable-declarative");
```

Comment:
you can just disable the declarative resourcemanagement in the configuration; explicit settings are honored even if the property is set.